### ISSUE TITLE
Feature request to disable window snapping on the top screen edge.

### ISSUE THREAD

**Issue by guzz46** [labwc/labwc#1765]
It would be great to have an option to disable window snapping on the top screen edge while retaining it everywhere else, kwin has this option, currently when you drag a window to the top of the screen it can trigger it to maximize or horizontal snap (depending on your config setting) when you just wanted to move it to the top of the screen, thanks.

**Comment by tokyo4j** [comment]
> kwin has this option

I'm trying KDE Plasma.
Looks like it supports tiling a window only to left/right edges and supports maximizing a window on top edge, right?
Then a possible solution to mimic this behavior would be adding something like `<snapping><disableVertical> [yes|no]` and still allowing to enable `<snapping><topMaximize>`.
Personally I'm fine with this feature to be added since I don't usually tile a window to top/bottom edges.

**Comment by guzz46** [comment]
> Looks like it supports tiling a window only to left/right edges and supports maximizing a window on top edge, right?

Yes, you can choose to disable the top screen edge so no maximizing or snapping happens when dragging a window to the top of the screen, while retaining the ability to snap windows to the left or right and quarter snapping on every corner too, personally I think its easier to maximize a window just by double clicking on the titlebar, which labwc already supports.

**Comment by Consolatis** [comment]
I wonder if something like this would be more future-proof:
```xml
<snapping>
	<topAction>SnapToEdge|Maximize|MaximizeGaps|None</topAction>
</snapping>
```
Additionally we can deprecate `topMaximize` over the next few releases.
`MaximizeGaps` (or some better name) would trigger `<action name="SnapToEdge direction="center" />` rather than Maximize.
